internal/renderer/render.go

```diff
@@ -218,11 +220,70 @@ func Render(appname string, composeFiles []string, settingsFile []string, env ma
 		}
 		configFiles = append(configFiles, composetypes.ConfigFile{Config: parsed})
 	}
+	return render(configFiles, finalEnv)
+}
+
+func render(configFiles []composetypes.ConfigFile, finalEnv map[string]string) (*composetypes.Config, error) {
 	rendered, err := loader.Load(composetypes.ConfigDetails{
-		WorkingDir:           ".",
-		ConfigFiles:          configFiles,
-		Environment:          finalEnv,
-		ErrOnMissingVariable: true,
+		WorkingDir:  ".",
+		ConfigFiles: configFiles,
+		Environment: finalEnv,
+	}, func(opts *loader.Options) {
+		opts.Interpolate.Substitute = substitute
 	})
-	return rendered, errors.Wrap(err, "failed to load Compose file")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to load Compose file")
+	}
+	processEnabled(rendered)
+	return rendered, nil
+}
+
+var delimiter = "\\$"
+var substitution = "[_a-z][._a-z0-9]*(?::?[-?][^}]*)?"
+
+var patternString = fmt.Sprintf(
+	"%s(?i:(?P<escaped>%s)|(?P<named>%s)|{(?P<braced>%s)}|(?P<invalid>))",
+	delimiter, delimiter, substitution, substitution,
+)
+
+var pattern = regexp.MustCompile(patternString)
+
+func substitute(template string, mapping composetemplate.Mapping) (string, error) {
+	return composetemplate.SubstituteWith(template, mapping, pattern, errorIfMissing)
+}
+
+func errorIfMissing(substitution string, mapping composetemplate.Mapping) (string, bool, error) {
+	value, found := mapping(substitution)
+	if !found {
+		return "", true, &composetemplate.InvalidTemplateError{
+			Template: fmt.Sprintf("required variable %s is missing a value", substitution),
+		}
+	}
+	return value, true, nil
+}
+
+func processEnabled(config *composetypes.Config) {
+	services := []composetypes.ServiceConfig{}
+	for _, service := range config.Services {
+		if service.Extras != nil {
+			if xEnabled, ok := service.Extras["x-enabled"]; ok && !isEnabled(xEnabled.(string)) {
+				continue
+			}
+		}
+		services = append(services, service)
+
+	}
+	config.Services = services
+}
+
+func isEnabled(e string) bool {
+	e = strings.ToLower(e)
+	switch {
+	case e == "", e == "0", e == "false":
+		return false
```

Comment:
I don't like the `case e == ""`, it means if someone uses `x-enabled` with empty value, it will be disabled:
```
node:
  x-enabled:
```
-> disabled
I think we should error out here. Enabling/Disabling a node should be explicit.

Reply:
Linking to #261